### DISCUSSION TITLE
Is there an event handler for irc.channel ?

### DISCUSSION THREAD

**Discussion by jrhoward** [savonet/liquidsoap#3733] [Q&A] [answered]
Hi all, I am looking at connecting my liquidsoap service to a privately managed irc server with 2.2.3 or 2.2.4. 
It works quite well and I can access the messages sent to the channel by accessing the returned function from `irc.channel` .
What I can't see is a how to add an event trigger that I can attached to the channel when a new message is sent.

Currently I have to call that function from other event triggers like on track and on metadata to inspect the messages sent to the channel.

Is there a way to trigger a function on message sent to channel ?

Any help appreciated , cheers !

**Comment by smimram** [comment, marked as answer]
You're right the current API is not very useful, I'll try to improve this soon!

**Reply by jrhoward** [reply]
Thanks for the quick response.

**Comment by jrhoward** [comment]
I built a simple irc  POC client in go that can sit beside the liquidsoap daemon and filter / forward commands via harbor so no rush on my side.
